### PR TITLE
IGNITE-17191 Consistency check should support cacheGroup as a param as well

### PR DESCRIPTION
**Pull request by anton-vinogradov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java [hunk 1]
```diff
@@ -117,73 +125,84 @@ protected VisorConsistencyRepairJob(VisorConsistencyRepairTaskArg arg, boolean d
...
-            StringBuilder sb = new StringBuilder();
-
-            sb.append("[node=").append(ignite.localNode());
-            sb.append(", cacheGroup=").append(grpCtx.cacheOrGroupName());
-            sb.append(", part=").append(p).append("]");
-
-            String statusKey = sb.toString();
+            String statusKey = "[node=" + ignite.localNode() + ", cacheGroup=" + grpCtx.cacheOrGroupName() + ", part=" + p + "]";
 
             if (VisorConsistencyStatusTask.MAP.putIfAbsent(statusKey, "0/" + part.fullSize()) != null)
```
Comment (ivandasch) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java:130]: This isn't an one-liner conditional branch, so it should be in curly braces

File: modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java [hunk 2]
```diff
@@ -292,13 +326,33 @@ private CacheConsistencyViolationEventListener(String name) {
...
+            private final IgniteCache<Object, Object> cache;
+
+            /** Keys. */
+            private final Set<Object> keys;
+
+            /**
+             * @param cache Cache.
+             * @param keys Keys.
+             */
+            public PerCacheBatch(IgniteCache<Object, Object> cache, Set<Object> keys) {
```
Comment (ivandasch) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java:352]: This constructor is always called with empty keys, seems that it should be changed a little bit : an empty hash set could be created during initialization

File: modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java (hunk 1 above)
Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java]: Fixed

File: modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java (hunk 2 above)
Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java:353]: Sure!

**Review by ivandasch — APPROVED** [review]

**Review by ivandasch — COMMENTED** [review]
I'd suggest to consider removing of the inner listener class. The job class itself can easily be a listener itself

**Pull request merged**